Marlin/src/module/settings.cpp

```diff
@@ -2573,7 +2579,7 @@ void MarlinSettings::reset() {
   //
   // TOUCH_SCREEN_CALIBRATION
   //
-  TERN_(TOUCH_SCREEN_CALIBRATION, touch_calibration.calibration_reset());
+  TERN_(TOUCH_SCREEN_CALIBRATION, touch_calibration.calibration_smart_reset());
```

Comment:
We don't need all this crazy logic. This is already solved in COLOR_UI. We just need to port the same logic to the other UIs: if you hold screen for `TOUCH_SCREEN_HOLD_TO_CALIBRATE_MS` (2.5s) on status screen, it will open touch calibration screen even if we have bad calibration values.

Reply:
Does it work in the EEPROM menu ? (Right now, the printer locks up in the printer "Reset EEPROM to factory default" menu, so it's not in the status screen).
Also, it's not intuitive or ergonomic. 

How is a user supposed to guess she should touch and hold to get calibration screen ?

Reply:
I can try to port the same logic to LVGL, but my goal is to improve user experience, and all "magic cheat code" are not good IMHO since you can't know them if you don't know the code perfectly. 
We must guide the user. So if we port the same COLOR_UI logic, we **must** display something (like "hold 3s to calibrate")  when the printer detects that the calibration is fucked up (or even better, automatically go to calibration menu since everything else will not work).

Reply:
> We don't need all this crazy logic. This is already solved in COLOR_UI. We just need to port the same logic to the other UIs

LVGL does not use `Marlin/Marlin/src/lcd/tft/touch.h`, and use `XPT2046` and `TouchCalibration` instead. Do you mean that you want to remove `TouchCalibration` class and make LVGL drop the XPT2046 code altogether to use `Touch` class instead ?

Reply:
No, `Touch` class is very specific to COLOR_UI. But we can mimic behaviours. For example: if LVGL_UI knows that a user is doing a EEPROM reset using menus, it can just send the user to the calibration screen, after a little delay. We don't need to change eeprom save/load logic. It should be done in all 3 ui? Sure, but each UI already do a lot of specific jobs and we won't add any crypt logic on load/save for calibration.

Reply:
For me, a reset should be a reset, and not a semi reset.
If the user send a M502, it should send to a M995, so it isn’t an issue for someone using serial.

It’s just an issue for someone doing reset using UI, so the UI should handle it.
And not adding UI logic inside eeprom load/save.

But it’s just my opinion.

Reply:
Ok, I'll do that and update the PR, you'll let me know.